raster/r.info/main.c

```diff
@@ -440,81 +461,149 @@ int main(int argc, char **argv)
+                    case PLAIN:
+                        fprintf(out, "min=%i\n", min);
+                        fprintf(out, "max=%i\n", max);
+                        break;
+                    case JSON:
+                        json_object_set_number(root_object, "min", min);
+                        json_object_set_number(root_object, "max", max);
+                        break;
+                    }
                 }
             }
             else {
                 DCELL min, max;
 
                 Rast_get_fp_range_min_max(&range, &min, &max);
                 if (Rast_is_d_null_value(&min)) {
-                    fprintf(out, "min=NULL\n");
-                    fprintf(out, "max=NULL\n");
+                    switch (format) {
+                    case PLAIN:
+                        fprintf(out, "min=NULL\n");
+                        fprintf(out, "max=NULL\n");
+                        break;
+                    case JSON:
+                        json_object_set_null(root_object, "min");
+                        json_object_set_null(root_object, "max");
+                        break;
+                    }
                 }
                 else {
-                    if (data_type == FCELL_TYPE) {
-                        fprintf(out, "min=%.7g\n", min);
-                        fprintf(out, "max=%.7g\n", max);
-                    }
-                    else {
-                        fprintf(out, "min=%.15g\n", min);
-                        fprintf(out, "max=%.15g\n", max);
+                    switch (format) {
+                    case PLAIN:
+                        if (data_type == FCELL_TYPE) {
+                            fprintf(out, "min=%.7g\n", min);
+                            fprintf(out, "max=%.7g\n", max);
+                        }
+                        else {
+                            fprintf(out, "min=%.15g\n", min);
+                            fprintf(out, "max=%.15g\n", max);
+                        }
+                        break;
+                    case JSON:
+                        json_object_set_number(root_object, "min", min);
+                        json_object_set_number(root_object, "max", max);
+                        break;
                     }
                 }
             }
         }
 
-        if (sflag->answer) {
+        if (sflag->answer || format == JSON) {
 
             if (!gflag->answer) {
+                grass_int64 total_cells =
+                    (grass_int64)cellhd.rows * cellhd.cols;
                 /* always report total number of cells */
-                fprintf(out, "cells=%" PRId64 "\n",
-                        (grass_int64)cellhd.rows * cellhd.cols);
+                switch (format) {
+                case PLAIN:
+                    fprintf(out, "cells=%" PRId64 "\n", total_cells);
+                    break;
+                case JSON:
+                    json_object_set_number(root_object, "cells", total_cells);
```

Comment:
Return an integer for the value of cells.

Reply:
Looking at this again, things like `"cells": 2025000.0,` in the output are a clear indicator we should format integers without decimal places.

Reply:
Changing format specifier to parson's default `%1.17g` fixed this.